scalike-jdbc-scala-fx/src/test/scala/fx/DatabaseSpec.scala

```diff
@@ -0,0 +1,52 @@
+package fx
+
+import java.sql.SQLException
+import scalikejdbc.*
+import scalikejdbc.DB.*
+
+class DatabaseSpec extends DatabaseSuite {
+
+  testFX("Read only operation executes properly") {
+    val names: Database[List[String]] = DB.readOnlyWithControl { implicit session =>
+      sql"select name from emp".map { rs => rs.string("name") }.list.apply()
+    }
+    assertFX(run(structured(names.join.toEither.isRight)) match {
+      case b: Boolean => b
+      case _ => false
+    })
+  }
+
+  testFX("Read only shift when tries to execute a non-read only operation") {
+    val names: Database[Int] = DB.readOnlyWithControl { implicit session =>
+      sql"update emp set name = 'Never changed in a readonly operation' where id = 1"
+        .update
+        .apply()
+    }
+    assertFX(run(structured(names.join.toEither.isLeft)) match {
+      case b: Boolean => false
+      case _ => true
+    })
```

Comment:
```suggestion
    assertFX(toEither(structured(DB.readOnlyWithControl { case given DBSession =>
      sql"update emp set name = 'Never changed in a readonly operation' where id = 1"
        .update
        .apply()
    }).join).isLeft)
```

If we remove the structured as suggested above, it can be:

```scala
assertFX(toEither(DB.readOnlyWithControl { case given DBSession =>
      sql"update emp set name = 'Never changed in a readonly operation' where id = 1"
        .update
        .apply()
    }).isLeft)
```

We can use `case given` instead of the deprecated `implicit`